article/app/pages/ArticleHtmlPage.scala

```diff
@@ -13,8 +13,11 @@ object ArticleHtmlPage extends HtmlPage[ArticlePage] {
     implicit val p: ArticlePage = page
 
     val (header, content): (Html, Html) = page.article match {
-      case article if (article.isPhotoEssay) => (photoEssayHeader(), immersiveBody())
-      case article if (article.isImmersive) => (immersiveHeader(), immersiveBody())
+      case article if article.isPhotoEssay => (photoEssayHeader(), immersiveBody())
+      case article if article.isImmersive => (immersiveHeader(), immersiveBody())
+      case _ if experiments.ActiveExperiments.isParticipating(experiments.Garnett) => (
```

Comment:
@NataliaLKB jsut realised this will break amp, since `articleBodyGarnett` doesn't have any amp stuff in it by design. i'm not sure how we determine that an article should be rendered for amp, it's not on the model at least. any thoughts about how this could ignore amp pages (and so send them back the to `articleBody`) once the switch is enabled?

Reply:
If you have access to request, you can can use the [isAmp](https://github.com/guardian/frontend/blob/master/common/app/implicits/Requests.scala#L24) check in the request.

```
case _ if (experiments.ActiveExperiments.isParticipating(experiments.Garnett) && !isAmp)
```

Reply:
Hope that helps! Let me know if thats not clear. Thanks 😄

Reply:
i _think_ it's available here https://github.com/guardian/frontend/blob/19ef10bc3c4d4fb5408f3fa57d35cc45fb0e379f/article/app/pages/ArticleHtmlPage.scala#L12 but `isAmp` doesn't resolve?

![screen shot 2017-12-06 at 14 21 03](https://user-images.githubusercontent.com/867233/33666057-bcdd214a-da90-11e7-8c0a-7ea5b54ae6fb.png)

Reply:
`isAmp` is coming from RichRequestHeader. Try importing that:

```
import common.RichRequestHeader
```

And then use `request.isAmp`

Reply:
bingo! thank you! ❤️